packages/components/src/list/experimental-list.tsx

```diff
@@ -0,0 +1,53 @@
+/**
+ * External dependencies
+ */
+import { Children, cloneElement, isValidElement } from '@wordpress/element';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+
+type ListProps = {
+	component?: string;
+} & React.HTMLAttributes< HTMLElement >;
+
+export const ExperimentalList: React.FC< ListProps > = ( {
+	children,
+	component = 'nav',
+	// Allow passing any other property overrides that are legal on an HTML element
+	...otherProps
+} ) => {
+	return (
+		<TransitionGroup
+			component={ component }
+			className="woocommerce-list"
+			{ ...otherProps }
+		>
+			{ /* Wrapping all children in a CSS Transition means no invalid props are passed to children and that anything can be animated. */ }
+			{
+				Children.map( children, ( child ) => {
+					if ( isValidElement( child ) ) {
+						const {
+							onExited,
+							in: inTransition,
+							enter,
+							exit,
+							...remainingProps
+						} = child.props;
+						return (
+							<CSSTransition
+								timeout={ 500 }
+								onExited={ onExited }
+								in={ inTransition }
+								enter={ enter }
+								exit={ exit }
+							>
+								{ cloneElement( child, { ...remainingProps } ) }
+							</CSSTransition>
+						);
+					}
+
+					return child;
+					// TODO - create a less restrictive type definition for children of react-transition-group. React.Children.map seems incompatible with the type expected by `children`.
```

Comment:
I don't think I can avoid a cast here without changing the type declaration of TransitionGroup. Happy to hear advice on this one. I could also write a type def instead of relying on the definitely typed one. I can see in their type defs they try to avoid children being something incompatible with the props they pass down into transition elements, but I don't think it covers things like text, booleans and nulls as covered by `React.ReactNode`.